.github/actions/publish-nuget/action.yml

```
@@ -60,7 +60,13 @@ runs:
         ${{ inputs.publish-version }}
         '@.Trim()
 
-        if ($version -eq 'USE_GITHUB_REF_NAME') { $version = $Env:GITHUB_REF_NAME.Trim().TrimStart("v") }
+        if ($version -eq 'USE_GITHUB_REF_NAME') {
+            $version = $Env:GITHUB_REF_NAME.Trim().TrimStart("v")
+            if (!($version -match '^\d+\.\d+\.\d+(-alpha\.\d+\.[a-z]{2,}-\d+)?$')) {
+                Write-Output "::error::$version is an invalid version number! The correct format is vM.N.O or vM.N.O-alpha.X.[issue-code]."
```

We could use angle brackets instead of the optional-looking square brackets then, @DAud-IcI:

```suggestion
                Write-Output "::error::`"v$version`" is invalid. The correct format is `"vM.N.O`" or `"vM.N.O-alpha.X.<issue-code>`"."
```

---

Please also change it in the documentation.

---

Agree.